app/services/processing_service.py

```
@@ -450,44 +523,282 @@ async def prepare_repository(self, repo_name) -> Tuple[Path, str]:
+            repo_url: URL of the git repository to clone
+            repo_path: Local path where the repository should be cloned
+            branch: Branch name to checkout (default: "main")
+            commit_hash: Specific commit hash to checkout (takes precedence over branch)
+
+        Returns:
+            List of documents loaded from the repository
+        """
+        import subprocess
+        import os
+        import shutil
+        from langchain_community.document_loaders.git import GitLoader
```

Ruff (0.14.4)

Redefinition of unused `GitLoader` from line 24

(F811)